File: book-backend/src/main/java/com/example/booklibrary/model/Book.java
package com.example.booklibrary.model;

import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.Instant;

public record Book(@MongoId String id,
                   String isbn,
                   String title,
                   String author,
                   Instant instant,
                   BookArt art,
                   String userId) {
    public Book(String isbn, String title, String author, BookArt art) {
        this(null,
                isbn,
                title,
                author,
                null,
                art, null);
    }


    public Book withUserId(String userId) {
        return new Book(null, isbn, title, author, Instant.now(), art, userId);
    }
}

File: book-frontend/src/model/Book.ts
export type Book = {
    id: string,
    isbn: string,
    title: string,
    author: string,
    art: "SOFTCOVER" | "EBOOK" | "HARDCOVER" | "AUDIOBOOK"
}

export type NewBook = {
    isbn: string,
    title: string,
    author: string,
    art: "SOFTCOVER" | "EBOOK" | "HARDCOVER" | "AUDIOBOOK"
}